airflow/api/common/experimental/delete_dag.py

```diff
@@ -40,6 +41,11 @@ def delete_dag(dag_id: str, keep_records_in_log: bool = True, session=None) -> i
     :return count of deleted dags
     """
     log.info("Deleting DAG: %s", dag_id)
+    running_tis = (
+        session.query(models.TaskInstance.state).filter(models.TaskInstance.state.in_(State.unfinished)).all()
```

Comment:
Doesn't this catch _any_ running TIs, not just TIs for _this DAG_?

Reply:
Additionally `.all()` is going to return all the rows. Better would be to use `.first()` (to only return a single row, or to use https://docs.sqlalchemy.org/en/13/orm/query.html?highlight=exists#sqlalchemy.orm.query.Query.exists (but note the caveat about mssql there)

Reply:
Ah, getting a fix now!

Reply:
my bad .. Too fast